network/android/src/main/java/com/capacitorjs/plugins/network/Network.java

```diff
@@ -59,26 +70,38 @@ public NetworkStatusChangeListener getStatusChangeListener() {
 
     /**
      * Get the current network information.
-     * @return NetworkInfo
+     * @return JSObject
      */
-    public NetworkInfo getNetworkStatus() {
-        return connectivityManager.getActiveNetworkInfo();
+    public JSObject getNetworkStatus() {
```

Comment:
We try not to use the JS related classes such as JSObject on the implementation classes as they are supposed to be used by native developers (if they want to).
Other plugins return a custom class with all the needed properties, or use separate calls for each value they want to get, so here you could use 2 functions, one for checking if connected or not and other for the connection type.
Or you can do what the iOS version of the plugin does, it just gets the connection string and based on the value, if it's "none" it returns false and true for the other values.

Reply:
Adjusted in new commit